articles/quickstart/webapp/nodejs-ea/01-login.md

```diff
@@ -0,0 +1,151 @@
+:::
+
+### Configure Logout URLs
+
+A logout URL is a URL in your application that Auth0 can return to after the user has been logged out of the authorization server. This is specified in the `returnTo` query parameter.
+
+The logout URL for your app must be whitelisted in the **Allowed Logout URLs** field in your [Application Settings](${manage_url}/#/applications/${account.clientId}/settings). If this field is not set, users will be unable to log out from the application and will get an error.
+
+::: note
+If you are testing your application locally, the logout URL you need to whitelist in the **Allowed Logout URLs** field is `http://localhost:3000`.
+:::
+
+## Integrate Auth0
+### Auth0 Application Keys
+
+You will need to retrieve some details for your application to communicate with Auth0. You can get these details from the [Application Settings](${manage_url}/#/applications/${account.clientId}/settings) section in the Auth0 dashboard.
+
+You need the following information:
+
+* **Domain**
+* **Client ID**
+* **Client Secret**
+
+### Express Middleware
+There are three middlewares required to integrate Auth0 in your application.
+
+- `express.urlencoded` - parses urlencoded payloads sent back from the Auth0 server.
+- `express-session` - simple session middleware for Express (other session middleware can be used as a replacement).
+- `express-openid-connect` - an [OIDC](/protocols/oidc) compliant authentication middleware.
+
+::: note
+To learn more about using middleware with Express view the [official documentation](https://expressjs.com/en/guide/using-middleware.html)
+:::
+
+::: note
+Secrets should never be checked into source control.  We recommend using environment variables in place of plain text.  You can use [Dotenv](https://github.com/motdotla/dotenv) to easily manage envrionment variables on your localhost.
+:::
+
+```js
+const express = require('express');
+const session = require('express-session');
+const { auth } = require('express-openid-connect');
+
+// parses urlencoded payloads sent back from the Auth0 server
+app.use(express.urlencoded({
+  extended: false
+}));
+
+// session middleware required to handle user sessions
+app.use(session({
+  secret: 'this should be a secret',
+  resave: true,
+  saveUninitialized: false
+}));
+
+// express-openid-connect.auth 
+app.use(auth({
+  // required - forces authentication for all routes
+  required: false,
+  // baseURL - There is where your /callback, /login & /logout URLs will be registered
+  baseURL: 'http://localhost:3000',
+  // issuerBaseURL - This is your Auth0 Domain found in your Auth0 Application Settings
+  issuerBaseURL: '${account.namespace}',
+  // clientID - This is your Auth0 Client ID found in your Auth0 Application Settings
+  clientID: '${account.clientId}',
+  // clientSecret - This is your Auth0 Client Secret found in your Auth0 Application Settings
+  clientSecret: '${account.clientSecret}'
+}));
+```
+
```

Comment:
This is probably a good place to call out that these values can be stored in `process.env` ([ref](https://github.com/auth0/express-openid-connect#getting-started)) and picked up automatically in the library. It would cut down on the amount of code that's here.

Reply:
I was going back and forth on the amount of code versus too much magic.  The library offers these options,  but I was worrying that it introduces too much magic too soon with too little understanding.  For example, `issuerBaseURL` if this doesn't change for the user, what is the point of using `process.env` for it?  

I'm not against moving these to `process.env`, as it definitely cleans up the amount of code needed, but it introduces an additional dependency for this tutorial.

Reply:
Apologies, I did not explain that very clearly. 

The `express-oidc-connect` library can pick up it's required config values from the environment automatically. [README has instructions on that](https://github.com/auth0/express-openid-connect#getting-started). So this QS would not get a new dependency, but it would be a bit of black magic and start to get into env vars. 

Since we're not using a client secret, configuring in the code is maybe not ideal but also not disastrous. All of these values are public (appear in the URL). So maybe leave as-is for simplicity and add a note like:

> For other configuration options and methods see the [docs](https://github.com/auth0/express-openid-connect/blob/master/API.md)